spec/classes/init_spec.rb

```diff
@@ -1459,17 +1272,49 @@
 
     context 'set to an invalid type (boolean)' do
       let(:params) { { :nagios_plugins_package_ensure => true } }
-      let(:facts) do
-        { :osfamily          => 'RedHat',
-          :lsbmajdistrelease => '6',
-        }
-      end
 
       it do
         expect {
           should contain_class('nrpe')
         }.to raise_error(Puppet::Error,/true is not a string\./)
       end
     end
+
+    describe 'with nrpe_package_provider parameter' do
+      context 'set to a valid string' do
+        let :params do
+          { 
+            :nrpe_package_provider  => 'sun',
+          }
+        end
+ 
+        it {
+          should contain_package('nrpe').with({
+            'provider' => 'sun',
+          })
+        }
+
+        it {
+          should contain_package('nagios-plugins').with({
+            'provider' => 'sun',
+          })
+        }
+      end
+
+      context 'set to an invalid type (non-string)' do
+        let :params do
+          { :nrpe_package           => 'nrpe',
+            :provider               => true,
+          }
+        end
+
+        it 'should fail' do
+          expect {
+            should contain_class('nrpe')
+          }.to raise_error(Puppet::Error)
```

Comment:
The error string that is returned must be tested to ensure that it errored in an expected way.